### ISSUE TITLE
BUG: end_time for Period with multiplied freq

### ISSUE THREAD

**Issue by pageecko** [pydata/pandas#11738]
end_time of a Period with multiplied freq is the square of the multiplication

```python
>>> pd.__version__
u'0.17.0'

>>> p = pd.Period('2015-08-01', freq='3D')
>>> p.end_time, p.to_timestamp(how='E')
(Timestamp('2015-08-09 23:59:59.999999999'), Timestamp('2015-08-03 00:00:00'))

>>> p = pd.Period('2015-08-01', freq='5D')
>>> p.end_time, p.to_timestamp(how='E')
(Timestamp('2015-08-25 23:59:59.999999999'), Timestamp('2015-08-05 00:00:00'))
```

**Comment by jreback** [comment]
yep, should be ``(p+1).start_time.value-1`` in ``src/period.pyx/Period/end_time``

**Comment by srib** [comment]
Please review and merge PR #11771.

**Comment by jreback** [comment]
closed by #11779
